File: dcp-solutions/2024-11-01/solution.php
```php
<?php
class Queue {
    private $stack1 = array();
    private $stack2 = array();

    public function enqueue($x) {
        array_push($this->stack1, $x);
    }

    public function dequeue() {
        if (empty($this->stack2)) {
            while (!empty($this->stack1)) {
                array_push($this->stack2, array_pop($this->stack1));
            }
        }
        return array_pop($this->stack2);
    }

    public function empty() {
        return empty($this->stack1) && empty($this->stack2);
    }
}
```

File: dcp-solutions/2024-11-01/solution.c
```c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    struct Stack* high;
    struct Stack* low;
} Queue;

struct Stack {
    int val;
    struct Stack* next;
};

Queue* new_queue() {
    Queue* q = (Queue*)malloc(sizeof(Queue));
    q->high = NULL;
    q->low = NULL;
    return q;
}

void enqueue(Queue* q, int x) {
    Stack* n = (Stack*)malloc(sizeof(Stack));
    n->val = x;
    n->next = NULL;

    if(q->low == NULL){
        q->high = n;
        q->low = n;
    } else {
        n->next = q->low;
        q->low->next = n;
        q->low = n;
    }
}

int dequeue(Queue* q) {
    int val = q->high->val;
    if(q->high == q->low){
        Stack* temp = q->high;
        q->high = NULL;
        q->low = NULL;
        free(temp);
    } else {
        Stack* top = q->high;
        q->high = q->high->next;
        top->next = NULL;
    }

    return val;
}
```